{"message":"`main` function has wrong type","code":{"code":"E0580","explanation":"The `main` function was incorrectly declared.\n\nErroneous code example:\n\n```compile_fail,E0580\nfn main(x: i32) { // error: main function has wrong type\n    println!(\"{}\", x);\n}\n```\n\nThe `main` function prototype should never take arguments.\nExample:\n\n```\nfn main() {\n    // your code\n}\n```\n\nIf you want to get command-line arguments, use `std::env::args`. To exit with a\nspecified exit code, use `std::process::exit`.\n"},"level":"error","spans":[{"file_name":"src/main.rs","byte_start":518,"byte_end":534,"line_start":30,"line_end":30,"column_start":1,"column_end":17,"is_primary":true,"text":[{"text":"unsafe fn main() {","highlight_start":1,"highlight_end":17}],"label":"expected normal fn, found unsafe fn","suggested_replacement":null,"suggestion_applicability":null,"expansion":null}],"children":[{"message":"expected fn pointer `fn()`\n   found fn pointer `unsafe fn()`","code":null,"level":"note","spans":[],"children":[],"rendered":null}],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror[E0580]\u001b[0m\u001b[0m\u001b[1m: `main` function has wrong type\u001b[0m\n\u001b[0m  \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m--> \u001b[0m\u001b[0msrc/main.rs:30:1\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m|\u001b[0m\n\u001b[0m\u001b[1m\u001b[38;5;12m30\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m|\u001b[0m\u001b[0m \u001b[0m\u001b[0munsafe fn main() {\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m| \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9m^^^^^^^^^^^^^^^^\u001b[0m\u001b[0m \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;9mexpected normal fn, found unsafe fn\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m|\u001b[0m\n\u001b[0m   \u001b[0m\u001b[0m\u001b[1m\u001b[38;5;12m= \u001b[0m\u001b[0m\u001b[1mnote\u001b[0m\u001b[0m: expected fn pointer `fn()`\u001b[0m\n\u001b[0m              found fn pointer `\u001b[0m\u001b[0m\u001b[1munsafe \u001b[0m\u001b[0mfn()`\u001b[0m\n\n"}
{"message":"aborting due to previous error","code":null,"level":"error","spans":[],"children":[],"rendered":"\u001b[0m\u001b[1m\u001b[38;5;9merror\u001b[0m\u001b[0m\u001b[1m: aborting due to previous error\u001b[0m\n\n"}
{"message":"For more information about this error, try `rustc --explain E0580`.","code":null,"level":"failure-note","spans":[],"children":[],"rendered":"\u001b[0m\u001b[1mFor more information about this error, try `rustc --explain E0580`.\u001b[0m\n"}
